src/stores/redis.store.ts

```diff
@@ -72,9 +72,12 @@ export const redisStore = (options: RedisStoreOptions) => {
   return <T>() => {
     const has = async (keys: string[]) => {
       const prefixedKeys = keys.map(key => prefixKey(key, prefix));
-      const count = await client.exists(...prefixedKeys);
-
-      return count === keys.length;
+      const exists = await Promise.all(
+        prefixedKeys.map(async prefixedKey => {
+          return (await client.exists(prefixedKey)) == 1;
+        })
+      );
+      return !exists.includes(false);
```

Comment:
I would use the [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method on the array instead of a map. That way the iteration stops if one item happens to be non-existent and thus prevent unnecessary calls to redis.

```typescript
const missing = await Promise.all(
  prefixedKeys.some(async prefixedKey => {
    return (await client.exists(prefixedKey)) !== 1;
  })
);

return !missing;
```

A potential issue with async functions inside of `map/some` is that all operations are processed simultaneously which could potentially overload redis but this is pure speculation at this point, I'm pretty confident redis is able to handle the load.

If we want to be on the safe side we can use a for loop which support async functions, resulting in sequential checks.

```typescript
for (const prefixedKey of prefixedKeys) {
  if (!(await client.exists(prefixedKey))) {
    return false;
  }
}

return true;
```

I noticed the use of double equals, which result in weak comparisons and can lead to weird bugs. It is recommended to always use the tripe equal operator. Looks like it is not enforced by our current eslint config. 

You can fix that by adding the `"eqeqeq": 2` rule to the `.eslintrc.js` file.

Reply:
.some does not appear to support async functions:

https://advancedweb.hu/how-to-use-async-functions-with-array-map-in-javascript/

https://advancedweb.hu/how-to-use-async-functions-with-array-some-and-every-in-javascript/

Reply:
Correct, you'd need to wrap it inside a `Promise.all` as you did for the `map`.